lib/response.js

```diff
@@ -847,7 +872,7 @@ res.redirect = function redirect(url) {
   });
 
   // Respond
-  this.statusCode = status;
+  this.status(status);
```

Comment:
Any reason for these changes? Since it's internal, presumably we already trust and can use the fast path to just assign?

Reply:
this part of the diff is lifted from the original work to land this in v4

i didn't really consider that we should drop it

the v4 support was largely bc we accept status as optional param (I think) in a few public methods


will take a closer look, open to your suggestion

Reply:
Sounds good, this case is reasonable. The other one is static but I don't have a strong opinion either way, go for it.

Reply:
I agree when we can easily have the fast path we should take it. In this case I doubt it is very meaningfully different but longer term it would be great to have better benchmarks in place to track how changes like this could impact the bigger perf picture.

Reply:
In this specific case, we still have varaitic inputs for this method, `res.redirect` which accept status. 

So doing the validation is useful when user provided status codes are in the mix.

Idk that we have a task tracking reducing the instances of methods having wonky signatures in v5.